internal/oci/uvm.go

```diff
@@ -434,5 +438,9 @@ func UpdateSpecFromOptions(s specs.Spec, opts *runhcsopts.Options) specs.Spec {
 		s.Annotations[AnnotationGPUVHDPath] = opts.GPUVHDPath
 	}
 
+	if _, ok := s.Annotations[AnnotationAssignedDeviceUtilityTool]; !ok && opts.DeviceUtilPath != "" {
+		s.Annotations[AnnotationAssignedDeviceUtilityTool] = opts.DeviceUtilPath
```

Comment:
Do we have a default path to search if nothing is specified in `DeviceUtilPath`, or is that required?

Reply:
It's a shim option so the default should be the value in the containerd.toml file. If that's not present, then we fail.

Reply:
Can we default to looking in the shim exe's directory or something? That seems like a reasonable default so this doesn't need to be configured by hand.

Reply:
I'm fine with that!

Reply:
Thinking about this some more, I don't think we should have an annotation for this tool at all. There's no use case where a user would want to use a different tool implementation instead, it should be treated as a detail of our package. I think we should just have the config file option, and the default fallback if that is not set.

Reply:
I think I included it as an annotation mostly for developer use. But I don't see that as necessary, so I'm fine with taking it out.